openedx/features/discounts/utils.py

```diff
@@ -134,5 +138,42 @@ def get_first_purchase_offer_banner_fragment(user, course):
                 span_close=HTML('</span>'),
                 div_close=HTML('</div>'),
                 strikeout_price=HTML(format_strikeout_price(user, course, check_for_discount=False)[0])
-            ))
+            )
+            return message_html
     return None
+
```

Comment:
@MatthewPiatetsky, @julianajlk, @cpennington: May I please trouble one of you to review this? I had to shift around the pieces a little in order to better isolate the data caching portion that I added.

Reply:
Right, it's still used by `course_home.py` I could have changed that call to use the new method, but then it'd be doing one more `CourseOverview` query than it would otherwise have to. I figured five lines of code was an acceptable tradeoff for that.